### ISSUE TITLE
PDF file dosn't generated correctly because URLs is doesn't decoded

### ISSUE THREAD

**Issue by eshcherbakov** [dotnet/docfx#1785]
### PDF file dosn't generated correctly because URLs in HTML is doesn't decoded.
PDF file contains only toc page.

### Minimal repro steps
- Create file `toc.md` with following content:
```
# [Введение](Intro.md)
# [Документ1](Документ1.md)
# [Документ2](Документ2.md)
# [Документ3](Документ3.md)
```
Files names (`Документ1.md`, `Документ2.md`, `Документ3.md`) contains cyrillic letters. 

- Run command `docfx pdf`;
- View file `_site_pdf\pdf\toc.html`. It contains following content:
```
<ul class="tocLevel3">
    <li><a class="normal" href="../articles/Intro.html">&#1042;&#1074;&#1077;&#1076;&#1077;&#1085;&#1080;&#1077;</a></li>
    <li><a class="normal" href="../articles/%D0%94%D0%BE%D0%BA%D1%83%D0%BC%D0%B5%D0%BD%D1%821.html">&#1044;&#1086;&#1082;&#1091;&#1084;&#1077;&#1085;&#1090;1</a></li>
    <li><a class="normal" href="../articles/%D0%94%D0%BE%D0%BA%D1%83%D0%BC%D0%B5%D0%BD%D1%822.html">&#1044;&#1086;&#1082;&#1091;&#1084;&#1077;&#1085;&#1090;2</a></li>
    <li><a class="normal" href="../articles/%D0%94%D0%BE%D0%BA%D1%83%D0%BC%D0%B5%D0%BD%D1%823.html">&#1044;&#1086;&#1082;&#1091;&#1084;&#1077;&#1085;&#1090;3</a></li>
</ul>
```
Here URLs is encoded, but physically files names is decoded:
```
Intro.html
Документ1.html
Документ2.html
Документ3.html
```
- View generated PDF-file. It contains only toc page and content of `Intro.html`.

### Expected result
Generated PDF-file contains toc page and content of files `Intro.html`, `Документ1.html`, `Документ2.html`, `Документ3.html`.

### Actual result
Generated PDF-file contains only toc and content of file `Intro.html`.

**Comment by vicancy** [comment]
https://github.com/wkhtmltopdf/wkhtmltopdf/pull/1769

**Comment by vicancy** [comment]
Hi @eshcherbakov I would suggest using simple file names, instead of cyrillic letters in file names, that will workaround current issue.
Current issue contains 2 inner issues: 1. Console std input in wkhtmltopdf does not support cyrillic letters 2. As https://github.com/wkhtmltopdf/wkhtmltopdf/pull/1769 describes, wkhtmltopdf does not correctly resolve links for cyrillic letters, although the PR claims the issue is fixed, I can still repro the issue in my local machine, so I am tracking the issue with them now.

There is not an easy fix for docfx to solve the issue, so for now, I would suggest using a simpler name, it will also be much more friendly URL when published as webpages

**Comment by eshcherbakov** [comment]
Hello, 
>  I would suggest using simple file names, instead of cyrillic letters in file names

Using of the Cyrillic letters is very important to us. Translating the names of files into English is a consuming task for technical writers.


> Console std input in wkhtmltopdf does not support cyrillic letters

Why?
1. I created a localized file, e.g. ТестоваяСтраница.html, with several content:
```html
<!DOCTYPE html>
<html>
   <head>
      <meta http-equiv="content-type" content="text/html; charset=utf-8">
   </head>
   <body>
      Это тестовая страница!
   </body>
</html>
```
2. Then I create the main file main.html containing:
```html
<!DOCTYPE html>
<html>
   <head>
      <meta http-equiv="content-type" content="text/html; charset=utf-8">
      <title>Title</title>
   </head>
   <body>
      <p>Ссылка на <a href="ТестоваяСтраница.html">ТестоваяСтраница</a></p>
   </body>
</html>
```
3. Convert: `wkhtmltopdf main.html ТестоваяСтраница.html test.pdf`.
4. File `test.pdf` is generated, the link is resolved correctly (see attached file [test.pdf](https://github.com/dotnet/docfx/files/1107860/test.pdf)).

Why do you encode the URL?
I need tag
```html
<a href="ТестоваяСтраница.html">
```
instead of this tag:
```html
<a href="%D0%A2%D0%B5%D1%81%D1%82%D0%BE%D0%B2%D0%B0%D1%8F%D0%A1%D1%82%D1%80%D0%B0%D0%BD%D0%B8%D1%86%D0%B0.html">
```

**Comment by vicancy** [comment]
Yes, it works when your locale is Russian, however when you change your "region and locales" to en-us, it fails. So it is not a general approach for 1.

For `href`, yes, it does work if the percent-encoded href gets decoded, however, I don't think it is a general practice neither, [href](https://www.w3schools.com/tags/att_a_href.asp) specifies the **URL** of the page the link goes to, and as an **URL**, it percent-encode cycrillic letter AFAIK. So I think it is a bug in wkhtmltopdf. Anyway, if they don't have any response unfortunately, I have to workaround the issue by decode `href`, although I think it is pretty tricky.

**Comment by eshcherbakov** [comment]
Thanks, @vicancy,

Can I write a plug-in that transferring the Cyrillic letters in the Latin file names and links?
E. g. 
`б` → `b`,
`т` → `t`,
`с` → `s` etc.

**Comment by vicancy** [comment]
@eshcherbakov you can add a yaml header to specify the output filename  https://github.com/dotnet/docfx/issues/1310
Is this acceptable?

**Comment by eshcherbakov** [comment]
@vicancy, It's OK.
But can I change the **directory** name by yaml header?

**Comment by vicancy** [comment]
@vwxyzh to confirm

**Comment by vicancy** [comment]
no change `directory` name is not supported. I will workaround the issue by decode percent encoded characters.

**Comment by yufeih** [comment]
Addressed in v2.73.0 with [a new PDF engine](https://dotnet.github.io/docfx/docs/pdf.html).
